### PR TITLE
Consolidated the OpenUX and Formik solution.

### PR DESCRIPTION
**Pull request by guzumba15**
Consolidated the OpenUX and Formik solution for each page currently available. Changes will come in another PR to make Yup work with might include minor changes to the TransactForm tag.

### CODE REVIEW

File: src/components/ContentContainer.tsx [hunk 1]
```diff
@@ -1,119 +1,183 @@
...
+                  </Page>
+                  <Page id="HousingInformation" label="Housing Information">
+                    <HousingInformation
+                      formValues={values}
+                      formTouched={touched}
+                      formErrors={errors}
+                      setFieldValue={setFieldValue}
+                    />
+                  </Page>
+                  <Page id="CoApplicantion" label="Co-application">
```
Comment (twilliams1128) [inline src/components/ContentContainer.tsx:134]: Typo: CoApplication

File: src/components/ContentContainer.tsx [hunk 2]
```diff
@@ -1,119 +1,183 @@
...
 
 export default function ContentContainer() {
   useEffect(getPreFillFormData, []);
 
   return (
     <div className="content-center z-10 relative top-40 pb-40">
       <div className="container content-center mx-auto">
-        <ProgressBar/>
+        <ProgressBar />
         <TransactForm
```
Comment (twilliams1128) [inline src/components/ContentContainer.tsx:89]: I've created a branch from this one that has the `TransactForm` removed. You can view it [here](https://github.com/guzumba15/PersonalLoan/tree/RemoveTransactForm), but basically you can replace the `TransactForm` tag with `Formik` and do the following in App.tsx. That should allow the previous validation you had previously implemented to work without much tweaking.
```
<TransactProvider>
      <PageProvider skipValidate={true}>
        <Header/>
        <ContentContainer />
      </PageProvider>
    </TransactProvider>
```

File: src/IFormProps.ts [hunk 3]
```diff
@@ -0,0 +1,6 @@
+export interface IFormProps {
```
Comment (twilliams1128) [inline src/IFormProps.ts:1]: Should we create a directory to hold different interfaces? So the structure would be src/interfaces/<interface>.ts

File: src/components/ContentContainer.tsx (hunk 1 above)
Comment (guzumba15) [reply src/components/ContentContainer.tsx]: Fixed now

File: src/components/ContentContainer.tsx (hunk 2 above)
Comment (guzumba15) [reply src/components/ContentContainer.tsx]: This is now integrated

File: src/IFormProps.ts (hunk 3 above)
Comment (guzumba15) [reply src/IFormProps.ts]: I ended up removing this whole IFormProps so we can revisit that when we add another interface. But yeah I agree

File: src/components/ContentContainer.tsx [hunk 4]
```diff
@@ -1,119 +1,86 @@
...
-    .catch(function (error) {
-      console.log("FormLoad error: " + error);
-    });
-  }
+  const pageContext = usePageContext();
 
-export default function ContentContainer() {
-  useEffect(getPreFillFormData, []);
+  useEffect(() => {
+    let pageIndex = pageContext.index ? pageContext.index : 0;
```
Comment (twilliams1128) [inline src/components/ContentContainer.tsx:25]: This can be shortened with the default operator: `pageIndex || 0`. I'm good with either

**Review by twilliams1128 — COMMENTED** [review]
Nicely done

**Review by twilliams1128 — APPROVED** [review]

**Pull request merged**